### PR TITLE
MOTECH-2741:Migrate reset password page into motech-auth

### PR DESCRIPTION
**Pull request by jhopen**

### CODE REVIEW

File: src/auth/reset_password/reset_password.directive.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+(function () {
+    'use strict';
+
+    angular.module('motech-auth')
+        .directive('motechResetpassword', ResetPasswordForm);
```
Comment (jredlarski) [inline src/auth/reset_password/reset_password.directive.js]: Rename directive to: 'motechResetPassword'.

File: src/auth/reset_password/reset_password.directive.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            scope:{},
+            templateUrl: '/auth/reset_password/reset_password.html',
+            controller: ResetFormController,
+            controllerAs: 'ResetCtrl',
+            link: Reset_PasswordFormDirective
+        };
+    }
+
+    ResetFormController.$inject = ['LoginModal', '$http', 'ResetPasswordModal'];
+    function ResetFormController(LoginModal, $http, ResetPasswordModal) {
```
Comment (jredlarski) [inline src/auth/reset_password/reset_password.directive.js]: Let's move this controller to a separate file reset-password.controller.js

File: src/auth/reset_password/reset_password.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        <div class="form-group">
+            <h4>{{ 'security.enterEmailQuestions' | translate }}</h4>
+        </div>
+        <div class="form-group">
+            <p>{{ 'security.enterEmailMsg' | translate }}</p>
+        </div>
+        <p ng-if="error" class="error">{{ error | translate }}</p>
+        <div class="form-group">
+            <label>{{ 'security.enterEmail' | translate }}</label>
+            <input id="auth.reset_password.email" class="col-sm-12 form-control" ng-model="email" name="email" type="email">
```
Comment (jredlarski) [inline src/auth/reset_password/reset_password.html]: Try not to use the `col-` classes for styling

File: src/auth/reset_password/reset_password.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<form id="auth.reset_password.form" name="form" class="inside form-horizontal" method="post" ng-submit="send()">
+    <div class="well3">
```
Comment (jredlarski) [inline src/auth/reset_password/reset_password.html]: This div is probably unnecessary and we don't use or style such class anywhere. Please remove it.

**Comment by jredlarski** [comment]
@jhopen I'm generally glad with your work but I would like you to fix a few things:

(1) Please change naming conventions for files from:
'reset_password.directive.js' to 'reset-password.directive.js' etc.
(2) Move the reset password directive controller logic to a separate file.

There is also a layout problem as it currently looks like this:
motech-reser-password-modal

I suggest styling it to make it look similarly to the login modal:
motech-sing-in-modal

So lets move the submit button to the bottom-right corner and the button for going back to the sign screen to the bottom-left corner. It can be done by removing the `<div class="form-group">` wrapper and leaving just:
`<input id="auth.reset_password.submit" type="submit" value="{{ 'security.sendReset' | translate }}"/>
        <button id="auth.reset_password.back" type="button" ng-click="back()" >{{ 'security.back' | translate }}</button>`
What is more, all other content should be separated from the borders of the modal window frame.

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: src/auth/reset_password/reset_password.modal.js [hunk 5]
```diff
@@ -10,7 +10,7 @@
         return {
             open: function () {
                 modal = ModalWindow(
-                    $compile('<motech-resetpassword />')($rootScope),
+                    $compile('<reset-password />')($rootScope),
                     $compile('<span>{{ "security.enterEmailQuestions" | translate}}</span>')($rootScope),
```
Comment (jredlarski) [inline src/auth/reset_password/reset_password.modal.js]: Let's change the modal title from "Cant sign in? Forgot your password?" to "Reset Instructions" as it was previously made in Motech. It doesn't look good to repeat the same words at the beginning of the modal content and in the modal title.

File: src/auth/reset_password/reset_password.html [hunk 6]
```diff
@@ -1,19 +1,15 @@
...
+
+    <p ng-if="error" class="error">{{ error | translate }}</p>
+
+    <div class="form-group">
+        <label>{{ 'security.enterEmail' | translate }}</label>
+        <input id="auth.reset_password.email" ng-model="email" name="email" type="email"/>
     </div>
-</form>
+
+    <input id="auth.reset_password.submit" class="btn btn-primary" type="submit" value="{{ 'security.sendReset' | translate }}"/>
```
Comment (jredlarski) [inline src/auth/reset_password/reset_password.html]: Change class to `button-primary`

File: src/auth/reset_password/reset_password.modal.js (hunk 5 above)
Comment (jredlarski) [inline src/auth/reset_password/reset_password.modal.js]: Let's change the modal title from "Cant sign in? Forgot your password?" to "Reset Instructions" as it was previously made in Motech. It doesn't look good to repeat the same words at the beginning of the modal content and in the modal title.

File: src/auth/reset_password/reset_password.modal.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        .service('ResetPasswordModal', ResetPasswordModal);
+
+    ResetPasswordModal.$inject = ['$compile', '$rootScope', 'ModalWindow'];
+    function ResetPasswordModal ($compile, $rootScope, ModalWindow) {
+        var modal;
+        return {
+            open: function () {
+                modal = ModalWindow(
+                    $compile('<reset-password />')($rootScope),
+                    $compile('<span>{{ "security.enterEmailQuestions" | translate}}</span>')($rootScope),
```
Comment (jredlarski) [inline src/auth/reset_password/reset_password.modal.js]: Let's change the modal title from "Cant sign in? Forgot your password?" to "Reset Instructions" as it was previously made in Motech. It doesn't look good to repeat the same words at the beginning of the modal content and in the modal title.

File: src/auth/reset_password/reset_password.html [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    <p>{{ 'security.enterEmailMsg' | translate }}</p>
+
+    <p ng-if="error" class="error">{{ error | translate }}</p>
+
+    <div class="form-group">
+        <label>{{ 'security.enterEmail' | translate }}</label>
+        <input id="auth.reset_password.email" ng-model="email" name="email" type="email"/>
+    </div>
+
+    <input id="auth.reset_password.submit" class="btn btn-primary" type="submit" value="{{ 'security.sendReset' | translate }}"/>
```
Comment (jredlarski) [inline src/auth/reset_password/reset_password.html]: Change class to `button-primary`

File: src/auth/reset_password/reset_password.directive.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+            scope: {},
+            controller: 'resetFormController',
+            templateUrl: '/auth/reset_password/reset_password.html',
+            link: Reset_PasswordFormDirective
+        };
+    }
+
+
+
+    function Reset_PasswordFormDirective (scope, element, attrs, controller) {
```
Comment (jredlarski) [inline src/auth/reset_password/reset_password.directive.js]: Remove redundant empty white lines. There should be a single empty line as a separator.

**Comment by jredlarski** [comment]
@jhopen It looks much better now. 

**(1)** I've made some inline comments which should be fixed.
**(2)** Also names of files don't follow our conventions so please change names for files from: `reset_password.directive.js` to `reset-password.directive.js` etc. Instead of using underscores `_` use dashes `-`.

**Comment by jhopen** [comment]
@jredlarski done

**Pull request merged**